src/classes/luaCoroutine.cpp

```diff
@@ -186,10 +190,29 @@ int LuaCoroutine::luaYield(lua_State *state) {
 }
 
 void LuaCoroutine::pause_execution() {
-    if (!done) lua_sethook(tState, terminate_hook, LUA_MASKCOUNT, 1); // force it to run next time
+    if (!done) lua_sethook(tState, pause_hook, LUA_MASKCOUNT, 1); // force it to run next time
 }
 
-void LuaCoroutine::terminate_hook(lua_State* tState,lua_Debug* dbg) {
+void LuaCoroutine::pause_hook(lua_State* tState,lua_Debug* dbg) {
    lua_sethook(tState, nullptr, NULL, NULL); // remove hook
    lua_yield(tState, 0);
 }
+
+void LuaCoroutine::interrupt_execution() {
+    if (!done) lua_sethook(tState, interrupt_hook, LUA_MASKCOUNT, 1); // force it to run next time
+}
+
+void LuaCoroutine::interrupt_hook(lua_State* tState,lua_Debug* dbg) {
+   lua_sethook(tState, nullptr, NULL, NULL); // remove hook
+   lua_pushstring(tState,"execution interrupted");
+   lua_error(tState);
+}
+
+void LuaCoroutine::kill() {
+    if (!done) lua_sethook(tState, terminate_hook, LUA_MASKCOUNT, 1); // force it to run next time
+}
+
+void LuaCoroutine::terminate_hook(lua_State* tState,lua_Debug* dbg) {
+   lua_pushstring(tState,"execution terminated");
+   lua_error(tState);
+}
```

Comment:
The hook in this case would remain. Meaning if we tried to use the state again it will error out immediately no?

Reply:
Yep, but isnt it hooked only to the running coroutine?

Reply:
It should be, be what if someone attempts to load another string or file into its state? Normally that should reset it.

Reply:
> It should be, but what if someone attempts to load another string or file into its state? Normally that should reset it.

I'll add the checks then.